app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/FullDetailsActivity.java

```diff
@@ -1326,6 +1346,29 @@ public boolean onMenuItemClick(MenuItem item) {
         }
     };
 
+    PopupMenu.OnMenuItemClickListener playWithMenuListener = new PopupMenu.OnMenuItemClickListener() {
+        @Override
+        public boolean onMenuItemClick(MenuItem item) {
+            switch (item.getItemId()) {
+
+                case R.id.play_with_vlc:
+                        userPreferences.getValue().set(UserPreferences.Companion.getVideoPlayer(), PreferredVideoPlayer.VLC);
```

Comment:
User preferences should **never** be changed without the user explicitly telling the application to do so. In this case you should find a way to set the player for the next play call without changing the setting.